src/mesh/mesh_smoother_vsmoother.C

```diff
@@ -44,4470 +50,73 @@ namespace libMesh
-
-          while (cells[i][nvert] >= 0)
-            {
-              x += R[cells[i][nvert]][0];
-              y += R[cells[i][nvert]][1];
-              if (_dim == 3)
-                z += R[cells[i][nvert]][2];
-              nvert++;
-            }
-
-          // adaptive function, cell based
-          afun[i] = 5.*(x+y+z);
-        }
-    }
-
-  else if (adp > 0)
-    {
-      for (dof_id_type i=0; i<_n_nodes; i++)
-        {
-  // This was an unused variable since Derek's first import of
-  // Larisa's code!?  Could something have been miscopied?
-/*
-          Real z = 0;
-          for (unsigned j=0; j<_dim; j++)
-            z += R[i][j];
-*/
-
-          // adaptive function, node based
-          afun[i] = 5*std::sin(R[i][0]);
-        }
-    }
-}
-
-
-
-// Preprocess mesh data and control smoothing/untangling iterations
-void VariationalMeshSmoother::full_smooth(Array2D<Real> & R,
-                                          const std::vector<int> & mask,
-                                          const Array2D<int> & cells,
-                                          const std::vector<int> & mcells,
-                                          const std::vector<int> & edges,
-                                          const std::vector<int> & hnodes,
-                                          Real w,
-                                          const std::vector<int> & iter,
-                                          int me,
-                                          const Array3D<Real> & H,
-                                          int adp,
-                                          int gr)
-{
-  // Control the amount of print statements in this function
-  int msglev = 1;
-
-  dof_id_type afun_size = 0;
-
-  // Adaptive function is on cells
-  if (adp < 0)
-    afun_size = _n_cells;
-
-  // Adaptive function is on nodes
-  else if (adp > 0)
-    afun_size = _n_nodes;
-
-  std::vector<Real> afun(afun_size);
-  std::vector<int> maskf(_n_nodes);
-  std::vector<Real> Gamma(_n_cells);
-
-  if (msglev >= 1)
-    _logfile << "N=" << _n_nodes
-             << " ncells=" << _n_cells
-             << " nedges=" << _n_hanging_edges
-             << std::endl;
-
-
-  // Boundary node counting
-  int NBN=0;
-  for (dof_id_type i=0; i<_n_nodes; i++)
-    if (mask[i] == 2 || mask[i] == 1)
-      NBN++;
-
-  if (NBN > 0)
-    {
-      if (msglev >= 1)
-        _logfile << "# of Boundary Nodes=" << NBN << std::endl;
-
-      NBN=0;
-      for (dof_id_type i=0; i<_n_nodes; i++)
-        if (mask[i] == 2)
-          NBN++;
-
-      if (msglev >= 1)
-        _logfile << "# of moving Boundary Nodes=" << NBN << std::endl;
-    }
-
-  for (dof_id_type i=0; i<_n_nodes; i++)
-    {
-      if ((mask[i]==1) || (mask[i]==2))
-        maskf[i] = 1;
-      else
-        maskf[i] = 0;
-    }
+  // Set this to something > 0 to add more quadrature points than the default
+  // rule that integrates order 2 * nq_points + 1 polynomials exactly.
```

Comment:
```suggestion
  // rule that integrates order 2 * fe_order + 1 polynomials exactly.
```

The version with `nq_points` is only valid in 1D